### PR TITLE
Add validation for password while updating system settings

### PR DESCRIPTION
**Pull request by vicpatel**
Admin > System settings for Password constraints,
- We need to validate that the provided min-length (when changing it) option is not less than 8; if it is it should error out
- Also, we should not allow negative numbers for any password constraints otherwise return an error.

Please take a look.

### CODE REVIEW

File: system/types/settings.go [hunk 1]
```diff
@@ -38,6 +39,7 @@ type (
 )
 
 const (
+	passwordMinLength        = 8
```
Comment (darh) [inline system/types/settings.go]: What is the reason that we're enforcing this?

File: tests/system/settings_test.go [hunk 2]
```diff
@@ -88,6 +88,21 @@ func TestSettingsUpdate_noPermissions(t *testing.T) {
 		End()
 }
 
+func TestSettingsUpdate_validation(t *testing.T) {
+	h := newHelper(t)
+	helpers.AllowMe(h, types.ComponentRbacResource(), "settings.manage")
+	helpers.AllowMe(h, types.ComponentRbacResource(), "settings.read")
+
+	h.apiInit().
```
Comment (darh) [inline tests/system/settings_test.go:97]: are there other cases and input variants we could test here?

**Review by darh — CHANGES_REQUESTED** [review]

File: tests/system/settings_test.go (hunk 2 above)
Comment (vicpatel) [reply tests/system/settings_test.go:97]: Yeah, we can I will update it.

File: system/types/settings.go (hunk 1 above)
Comment (vicpatel) [reply system/types/settings.go]: We have a default min-length for password is 8.

**Pull request merged**